frontend/src/views/admin/matching/match-actions/edit-applicant-note.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { ApplicantSummary } from "../types";
+import { Form, Modal, Button } from "react-bootstrap";
+import { upsertNote } from "../actions";
+import { useThunkDispatch } from "../../../../libs/thunk-dispatch";
+
+/**
+ * A modal window allowing users to view and edit notes for an applicant.
+ */
+export function ApplicantNoteModal({
+    applicantSummary,
+    show,
+    setShow,
+}: {
+    applicantSummary: ApplicantSummary;
+    show: boolean;
+    setShow: (show: boolean) => void;
+}) {
+    const dispatch = useThunkDispatch();
+    const [noteTemp, setNoteTemp] = React.useState(applicantSummary.note || "");
+
+    const updateApplicantNote = React.useCallback(() => {
```

Comment:
Why are you doing `useCallback` here? You're not getting the benefits of it on L60, since you're dynamically creating a surrounding callback.